### PR TITLE
Api

### PR DESCRIPTION
**Pull request by allistergg**

### CODE REVIEW

File: customer-node/routes/flights.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  const datere = /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/;
+  if (!date.match(datere)) {
+    return res.status(400).json({ message: 'Invalid Date' });
+  }
+  if (source.length !== 3 || destination.length !== 3) {
+    return res.status(400).json({ message: 'Invalid Airport' });
+  }
+  try {
+    const flights = await getFlights(source, destination, date);
```
Comment (AlexChavez235) [inline customer-node/routes/flights.js:24]: What if the airportCodes are 3 characters but the airport does not exist?

File: customer-node/service/customerService.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const { findCustomer, createCustomer, updateCustomerData } = require('../data_access/customerda');
+
+const getCustomer = (customerId) => findCustomer(customerId);
+
+const addCustomer = (customer) => createCustomer(customer);
+
+const updateCustomer = (customer) => updateCustomerData(customer);
```
Comment (AlexChavez235) [inline customer-node/service/customerService.js:7]: What happens when you try to update a customer that does not exist?

File: customer-node/data_access/flightda.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+const findFlights = async (source, destination, date) => {
+  try {
+    const flights = await knex('Flight').where({
+      sourceAirport: source,
+      destinationAirport: destination,
+      departureDate: date,
+    });
+    return flights;
+  } catch (err) {
+    return [];
```
Comment (AlexChavez235) [inline customer-node/data_access/flightda.js:12]: How to differentiate between an error and just not having any flights with the 3 fields?

File: customer-node/service/flightsService.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const { findFlights } = require('../data_access/flightda');
+
+const getFlights = (source, destination, date) => findFlights(source, destination, date);
```
Comment (AlexChavez235) [inline customer-node/service/flightsService.js:3]: Maybe check if source and destination exist in DB before querying for flights?

File: customer-node/service/ticketsService.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+const {
+  findTicketsByCustomer, createTicket, findTicketById, deleteTicket,
+} = require('../data_access/ticketda');
+
+const getTicketsByCustomerId = (customerId) => findTicketsByCustomer(customerId);
+
+const addTicket = (ticket) => createTicket(ticket);
```
Comment (AlexChavez235) [inline customer-node/service/ticketsService.js:7]: Make sure to decrease seatsAvailable in flights here

File: customer-node/service/ticketsService.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+const {
+  findTicketsByCustomer, createTicket, findTicketById, deleteTicket,
+} = require('../data_access/ticketda');
+
+const getTicketsByCustomerId = (customerId) => findTicketsByCustomer(customerId);
```
Comment (AlexChavez235) [inline customer-node/service/ticketsService.js:5]: Check if a customer with such an Id exists before querying for tickets

File: customer-node/service/ticketsService.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+const addTicket = (ticket) => createTicket(ticket);
+
+const cancelTicket = async (customerId, ticketId) => {
+  const ticket = await findTicketById(ticketId);
+  if (ticket[0].customerId !== customerId) {
```
Comment (AlexChavez235) [inline customer-node/service/ticketsService.js:11]: Check if ticket is null before doing this comparison

File: customer-node/service/ticketsService.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    const err = new Error('Ticket does not belong to customer');
+    err.status = 401;
+    throw err;
+  }
```
Comment (AlexChavez235) [inline customer-node/service/ticketsService.js:15]: Make sure to increment seats available in flight as well

File: customer-node/routes/flights.js (hunk 1 above)
Comment (allistergg) [reply customer-node/routes/flights.js:24]: changed from 200 empty array to 400 error

File: customer-node/service/customerService.js (hunk 2 above)
Comment (allistergg) [reply customer-node/service/customerService.js:7]: changed from no content to 404 error

File: customer-node/data_access/flightda.js (hunk 3 above)
Comment (allistergg) [reply customer-node/data_access/flightda.js:12]: added 400 errors

File: customer-node/service/flightsService.js (hunk 4 above)
Comment (allistergg) [reply customer-node/service/flightsService.js:3]: done

File: customer-node/service/ticketsService.js (hunk 6 above)
Comment (allistergg) [reply customer-node/service/ticketsService.js:5]: done

File: customer-node/service/ticketsService.js (hunk 7 above)
Comment (allistergg) [reply customer-node/service/ticketsService.js:11]: done

File: customer-node/service/ticketsService.js (hunk 5 above)
Comment (allistergg) [reply customer-node/service/ticketsService.js]: done

File: customer-node/service/ticketsService.js (hunk 8 above)
Comment (allistergg) [reply customer-node/service/ticketsService.js:39]: done

**Pull request merged**